doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,616 @@
+```
+it -> 10\n5\n
+was -> 10\n
+the -> 14\n7\n
+best -> 1\n
+worst -> 1\n
+of -> 4\n4\n
+times -> 2\n
+age -> 2\n
+wisdom -> 1\n
+foolishness -> 1\n
+```
+During each job that is run, each distinct datum in Pachyderm will put data in an output file.  If the file shares a name with the files from other datums,  they're merged after processing. This will happen during the appropriately-named _merge_ stage after your pipeline runs. You should not count on the data appearing in a particular order, or the data in a file in an output repo from a particular datum containing the data from the processing of any other datums.  You won't see it in the file until all datums have been processed and the merge is complete, after that pipeline is finished.
+
+What happens if we delete `md.txt`?  The "wordcount" repo would go back to its condition with just `tts.txt`.
+```
+it -> 10\n
+was -> 10\n
+the -> 14\n
+best -> 1\n
+worst -> 1\n
+of -> 4\n
+times -> 2\n
+age -> 2\n
+wisdom -> 1\n
+foolishness -> 1\n
+```
+What if didn't delete `md.txt`; we appended to it?  Then we'd see the appropriate counts change only on the lines of the files affected by `md.txt`; the counts for `tts.txt` would not change. Let's say we append the second paragraph to `md.txt`:
+```
+There now is your insular city of the Manhattoes, belted round by
+wharves as Indian isles by coral reefs—commerce surrounds it with her
+surf. Right and left, the streets take you waterward. Its extreme
+downtown is the battery, where that noble mole is washed by waves, and
+cooled by breezes, which a few hours previous were out of sight of
+land. Look at the crowds of water-gazers there.
+```
+The "wordcount" repo might now look like this. (We're not using stemmed parser, and "it" is a different word than "its")
+```
+it -> 10\n6\n
+was -> 10\n
+the -> 14\n11\n
+best -> 1\n
+worst -> 1\n
+of -> 4\n8\n
+times -> 2\n
+age -> 2\n
+wisdom -> 1\n
+foolishness -> 1\n
+```
+Pachyderm is smart enough to keep track of what changes to what datums affect what downstream results, and only reprocesses and re-merges as needed.
+
+To summarize, 
+
+- If you overwrite an input datum in pachyderm with new data, each output datum that can trace to that input datum in downstream pipelines will get updated 
+- If your downstream pipeline processes multiple input datums, putting the result a single file, adding or removing an input datum will only remove its effect from that file.  The effect of the other datums will still be seen in that file.
+
+You can see this in action in the [word count example](https://github.com/pachyderm/pachyderm/tree/master/examples/word_count) in the Pachyderm git repo.
+
+## Deduplication of output data
+
+Data is not currently deduplicated globally across output repos for performance reasons, only locally, within that repo. If you copy data from an output repo to an input repo, it will be deduplicated globally.
```

Comment:
Not sure if this is accurate or if we actually support this right now. lets chat with eng.

Reply:
I removed this so we can ship this thing, and will add it back if we confirm.